### PR TITLE
fix(macros/LearnSidebar): adapt sidebar navigation

### LINKED ISSUE #5164: Issue with "CSS building blocks": Assessments missing in the navigation
MDN URL: https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks

#### What information was incorrect, unhelpful, or incomplete?
In the navigation (Related Topics) on the left, the assessments are not displayed for the ["CSS building blocks" module](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks).

#### Specific section or headline?
Related Topics navigation on the left.

#### What did you expect to see?
Until now, the assessments were always displayed in the navigation on the left. For [this module](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks), I assumed that there were no assessments and almost overlooked them.

#### Did you test this? If so, how?
Assessments just on the [overview section of the page](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks#assessments), not in the navigation.
The ["CSS layout" module](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout) navigation for the assessments is also missing - additionally, the heading "Assessments" is missing on the page in general.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/learn/css/building_blocks`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/css/building_blocks/index.md
* Last commit: https://github.com/mdn/content/commit/07453ac2b659a397b0fee921fa71db5488e9ca62
* Document last modified: 2021-10-08T14:43:33.000Z

</details>

### PR DESCRIPTION
**Pull request by glmvc**
## Summary

Fixes #5164 and adjusts some minor "typos" (upper and lower case letters)

### Problem

- For the [CSS building blocks module](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks) the assessments are missing in the sidebar.
  - When I was learning, I assumed this module had no assessments because I always paid attention to the sidebar - so I almost overlooked them
- In the [CSS layout module](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout) "Assessment: " is missing before the assessment title in the sidebar.
  - for the other modules it is also like that (I think it is so to easily distinguish the asssessments from "normal" articles)
- In the [CSS first steps module](https://developer.mozilla.org/en-US/docs/Learn/CSS/First_steps) the assessment got a new title (see this [pr in mdn/content](https://github.com/mdn/content/pull/13781))
- Sometimes the capitalization was different from the article titles themselves
  - to keep it consistent

### Solution

- Updated the HTML template by adding the missing assessments
  - in the `text` dictionary I've also added the matching string identifiers
  - because I didn't know exactly how translations are handled, I left them all on the English translation
- Added "Assessment: " before the assessment title
  - where a translation was present, I also adapted it by comparing and copying previous translations of the particular language for "Assessment"
- Adaptet the string identifiers and translations for the new assessment title
- Adaptet the capitalization to fit the article titles

> For some translations I used [Deepl Translate](https://www.deepl.com/) and compared it with the current translation to check if my changes fit correctly. But as I said, I do not know how translations are handled here...

---

## How did you test this change?

Because I'm relatively new to all of this, I didn't know exactly how to test this. Maybe someone can explain this to me please.
Of course I will then test the whole thing!

### CODE REVIEW

**Comment by glmvc** [comment]
> ## How did you test this change?
> Because I'm relatively new to all of this, I didn't know exactly how to test this. Maybe someone can explain this to me please. Of course I will then test the whole thing!

I managed how to setup yarn and test my changes.
I have checked the changed strings in the sidebar and it all seems to work.

**Comment by cw118** [comment]
Hi @glmvc, I was just wondering if there's a reason why the French locale (lines 1527-1551) sidebar wasn't updated here?

On another note, signed commits are a requirement on Yari, I recommend reading through the [commit signatures module on Github docs](https://github.com/mdn/yari/pull/5828/files#diff-f42bbd8a5a864e6d8d40c7f6fc6654d123d9697244248a6e13556fe975d53aa7L1527-L1551) for more information :)

**Comment by glmvc** [comment]
Hey @cw118, I didn't even notice that most of it is missing for French, since I was basically just working with "Find and Replace". Maybe someone else knows more? Or should I add to it and have someone who knows French in another PR do the translations?

By the way, thanks for the note about the signed commits - I'll take a look at it :)

**Comment by cw118** [comment]
@glmvc after giving it some more thought, I think it could be easier if we (the l10n-fr team) fix the French sidebar in our own PR just because there are so many links/translations missing. Otherwise I wouldn't be against adding the missing ones in English here and translating separately later on (or even commenting on the file with suggested translations here). 

Let me know what you think :)

**Comment by glmvc** [comment]
@cw118 i also think it would be better practice if we file a separate pr for this :) as you said, otherwise it gets cluttered / messy...

we should definitely **create an issue** so that it won't be forgotten - should I do that or are you more familiar with the french team / yari in general?

thanks in advance!

**Comment by cw118** [comment]
A separate PR it is then :) 

> we should definitely create an issue so that it won't be forgotten - should I do that or are you more familiar with the french team / yari in general?

To be honest, French translations for most/all sidebars are quite out of date. They're not necessarily a priority (as far as I know at least) but I guess making an issue could serve as a reminder

**Comment by glmvc** [comment]
@cw118 I've created an issue (#5994).

I have a question about the commit signatures:
As far as I have researched, it is enough if the latest commit has a signature, right? So should I create another commit with a signature?
Or am I misinformed? If so, is it possible to add a signature to my commits in this pr?

Thank you in advance for your help! :)

**Comment by cw118** [comment]
Thank you for making the issue!

> As far as I have researched, it is enough if the latest commit has a signature, right? So should I create another commit with a signature?

Just to be sure, I tested this on my own repository, and I believe **all** of your commits need to be signed. 

<details>
<summary>Here's one way to do/fix this via interactive rebase:</summary>
<br>

1. First, you'll want to get the hash of the commit that comes before your two commits on this branch. You can do this with `git log --oneline` *(press the `q` key to "exit")* which should show something like this:

  ```
  206fc14d9 (HEAD -> sidebar-fix, origin/sidebar-fix) Fix new assessment title
  ab39ce7f0 Fix issue 5164: Adapt sidebar navigation
  874ccffab (tag: v0.9.60, origin/main) Make it possible to use an absolute URL when providing a redirect (#5814)
  05e7a8807 (tag: v0.9.59) fix(active-banner): make "MDN Plus" a link in the banner
  ```
2. Since the two commits you want to sign are `206fc14d9` (adapt sidebar) and `ab39ce7f0` (fix assessment title), the commit hash to use for the interactive rebase is **`874ccffab`** (use absolute URL for redirect). This means you can use the command:  
  ```
  git rebase --exec "git commit --amend --no-edit -n -S" -i 874ccffab
  ```
  Grosso modo, the line above will start an [interactive rebase (the `-i` option)](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase) listing all commits that come after the one with `874ccffab` as a hash, and tells git to sign those commits (the `-S` option) without modifying their commit messages (the `--no-edit` option). It should open a file in your editor (depends on what you use) showing something like the following:
  ```
  pick ab39ce7f0 Fix issue 5164: Adapt sidebar navigation
  exec git commit --amend --no-edit -n -S
  pick 206fc14d9 Fix new assessment title
  exec git commit --amend --no-edit -n -S

  # Rebase 874ccffab..206fc14d9 onto 874ccffab (4 commands)
  ```
3. If you use an editor like VS Code and the rebase file (likely named `git-rebase-todo` in VS Code) looks right, you can close the file and go back to the command line. Enter your password to sign the commits if you're asked for it, git should execute the rebase and sign your two commits (`adapt sidebar` and `fix assessment title`) after that (if it worked you should see something like `Successfully rebased and updated …`).
4. Now you need to push to `origin`. A [force push](https://www.git-tower.com/learn/git/faq/git-force-push) is necessary here because after rebasing, the commits you have on your local machine (signed) are "different" from the ones on the remote (unsigned), so you need to "replace" the commits on the remote with the ones on your local machine. You can use this command:
  ```
  git push origin sidebar-fix --force-with-lease
  ```
  [This article](https://itnext.io/git-force-vs-force-with-lease-9d0e753e8c41) has a great explanation of using `--force` (or `-f`) vs `--force-with-lease` if you're curious :)

</details>

And (hopefully) your commits in this pull request will be signed! I hope the (very long :/) explanation helps :)

**Comment by glmvc** [comment]
@cw118 thank you very much for the detailed explanation - appreciate it! 🎉 
I see, the interactive rebase is quite advanced (for me at least)...
but i managed to resolve the conflicts here on GitHub and the merge commit got it's signature.

now does that meet the merging requirements, or does the rebase need to be done anyways?

**Comment by cw118** [comment]
> I see, the interactive rebase is quite advanced (for me at least)...
but i managed to resolve the conflicts here on GitHub and the merge commit got it's signature.
> 
> now does that meet the merging requirements, or does the rebase need to be done anyways?

Based on my own tests and the "merging is blocked" message, **all** commits need to be signed for the requirement to be met.  A maintainer can, however, "disregard" this and merge even if a PR has unsigned commits — and since an interactive rebase is quite advanced, I probably wouldn't worry too much about signing your first two commits here. Just keep the commit signature requirement in mind in the future :)

**Comment by glmvc** [comment]
@cw118 okay, now I understand :) Many thanks for the help and explanations!

**Comment by wbamberg** [comment]
@glmvc , for places where you've changed translated text it might be a good idea to tag the maintainers of those locales to check. @SphinxKnight , could you help to reach translators to take a look at this PR?

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks @glmvc ! As noted I think we should use sentence case everywhere (i.e. "Multimedia and embedding"), and for text you've changed that's not en-US, it's probably worth getting the localizers to have a look.

**Comment by schalkneethling** [comment]
@mdn/localization-team-leads Please have a look at the change proposed here for your relevant language. Thank you.

**Comment by SphinxKnight** [comment]
Regarding `fr`, nothing is changed here (subsumed by https://github.com/mdn/translated-content/issues/5135 thanks to @cw118 )

**Review by myshov — APPROVED** [review]
👍 for @mdn/yari-content-ru

File: kumascript/macros/LearnSidebar.ejs
Comment (glmvc) [reply kumascript/macros/LearnSidebar.ejs]: I agree. I looked over it again and tried to make the spelling consistent in sentence case (as with most article titles).
I've also adapted the string identifiers to fit the article titles better.
Hope the new changes fit well

**Comment by glmvc** [comment]
Hi Team,

I was wondering what is actually going on with the other languages that MDN offers.
The translations for the following languages that are present here:
- English (US) of course
- German
- Portuguese (Brazilian)
- Russian
- Chinese (both simplified and traditional)
- French (Note [issue #5135](https://github.com/mdn/translated-content/issues/5135))
- Japanese

But what about the others:
- Spanish
- Polish
- Korean

Are these missing ones tracked somewhere or are these translations missing here as well? Is there a translation team for these languages? @SphinxKnight

**Comment by glmvc** [comment]
#### I tested all additional changes (new commits) by setting up yarn and checking the sidebar in the learning area (textually and if the links work) ✔️ 

> Note: These changes affect only the "Getting started with the Web", "Learn HTML", and "Learn CSS" modules (through my previous prs and my learning progress on mdn, I only have the certain overview of these modules).
>If desired, I would make such adjustments (if needed) to the sidebars in future prs as my (learning) progress on mdn gets progressed enough (e.g. Learn JavaScript, Web forms or Accessibility modules).

**Comment by hochan222** [comment]
Regarding `ko`, nothing is changed here.

In `ko-locale`, there is currently no related code, so the side menu is provided in `en-us`. I will create an issue later and add it.

**Comment by caugner** [comment]
Friendly reminder that this PR is waiting for approval by:

- @mdn/yari-content-pt-br
- @mdn/yari-content-zh (both `zh-CN` and `zh-TW`)
- @mdn/yari-content-ja

File: kumascript/macros/LearnSidebar.ejs
Comment (caugner) [inline kumascript/macros/LearnSidebar.ejs:833]: @myshov Based on https://developer.mozilla.org/ru/docs/Learn/CSS/First_steps/Styling_a_biography_page, the translation should remain the same:

```suggestion
        'Assessment_Styling_a_biography_page': 'Использование ваших новых знаний',
```

**Review by wbamberg — APPROVED** [review]
👍 thanks @glmvc , this looks great to me.

I'm happy to merge, subject to translators being happy with translations. But also I think if we don't hear back soon from everyone, we should merge this and deal with any issues as a follow-up.

It did occur to me that we should probably use only lowercase for the identifiers, like:

> 'complete_beginners_start_here': 'Complete beginners start here!',

But I don't want to block this PR any more on issues like this :).

**Comment by irvin** [comment]
ok for both zh

**Review by caugner — APPROVED** [review]

**Comment by caugner** [comment]
Thank you @glmvc! 🎉

**Pull request merged**
